apps/web/src/components/templates/TemplateEditorProvider.tsx

```diff
@@ -1,132 +1,315 @@
+            ...item,
+            ...(!item.replyCallback && {
+              replyCallback: {
+                active: false,
+                url: '',
+              },
+            }),
+            template: {
+              ...item.template,
+              layoutId: item.template._layoutId ?? '',
+              preheader: item.template.preheader ?? '',
+              content: item.template.content,
+              ...(item.template?.contentType === 'customHtml' && {
+                htmlContent: item.template.content as string,
+                content: [],
+              }),
+            },
+          };
+        }
+        if (item.template.type === StepTypeEnum.IN_APP) {
+          return {
+            ...item,
+            template: {
+              ...item.template,
+              feedId: item.template._feedId ?? '',
+              actor: item.template.actor?.type
+                ? item.template.actor
+                : {
+                    type: ActorTypeEnum.NONE,
+                    data: null,
+                  },
+              enableAvatar: item.template.actor?.type && item.template.actor.type !== ActorTypeEnum.NONE ? true : false,
+              cta: {
+                data: item.template.cta?.data ?? { url: '' },
+                type: ChannelCTATypeEnum.REDIRECT,
+                action: item.template.cta?.action ?? '',
+              },
+            },
+          };
+        }
+
+        return item;
+      });
+
+      reset(formValues);
+      setTrigger(template.triggers[0]);
+    }
+  }, [isDirtyForm, template]);
+
+  const onSubmit = useCallback(
+    async (data: IForm, { onCreateSuccess } = {}) => {
+      let stepsToSave = data.steps;
+
+      stepsToSave = stepsToSave.map((step: IStepEntity) => {
+        if (step.template.type === StepTypeEnum.EMAIL && step.template.contentType === 'customHtml') {
+          step.template.content = step.template.htmlContent as string;
+        }
+
+        if (step.template.type === StepTypeEnum.IN_APP) {
+          if (!step.template.enableAvatar) {
+            step.template.actor = {
+              type: ActorTypeEnum.NONE,
+              data: null,
+            };
+          }
+
+          delete step.template.enableAvatar;
+        }
+
+        return step;
+      });
+
+      const payloadToCreate: ICreateNotificationTemplateDto = {
+        name: data.name,
+        notificationGroupId: data.notificationGroupId,
+        description: data.description !== '' ? data.description : undefined,
+        tags: data.tags,
+        critical: data.critical,
+        preferenceSettings: data.preferenceSettings,
+        steps: stepsToSave,
+      };
+
+      try {
+        if (editMode) {
+          const response = await updateNotificationTemplate({
+            id: templateId,
+            data: {
+              ...payloadToCreate,
+              identifier: data.identifier,
+            },
+          });
+          setTrigger(response.triggers[0]);
+          reset(data);
```

Comment:
What's the reason that here we reset the data but in line 217 we reset the `payloadToCreate`? I would expect `data` to be reset in both. What am I missing?

Reply:
not sure, it was like that before... I feel it doesn't have an impact on the form at all because after creation we redirect the user to the `/templates` route